src/pamiq_core/model/interface.py

```diff
@@ -63,11 +66,11 @@ def inference_model(self) -> InferenceModel:
         return self._inference_model
 
     @abstractmethod
-    def _create_inference_model(self) -> InferenceModel:
+    def _create_inference_model(self) -> T:
         """Create inference model.
 
         Returns:
-            InferenceModel: A model to infer.
+            T: A model to infer.
```

Comment:
doc stringに型情報を載せるとこのように二重管理が発生する。適当に書いておくのが良い
```suggestion
            InferenceModel.
```